Packs/CommonScripts/Scripts/FormatURL/FormatURL.py

```diff
@@ -104,16 +105,65 @@ def replace_protocol(url_: str) -> str:
     return url_
 
 
-def remove_brackets_from_end_of_url(url_: str) -> str:
+def remove_special_chars_from_start_and_end_of_url(url_: str) -> str:
     """
-    Removes square brackets from the end of URL if there are any.
+    Removes brackets and quotes characters from the beginning and the end of the URL if there are any.
     Args:
-        url_ (str): URL to remove the brackets from.
-
+        url_ (str): URL to remove the special characters from.
     Returns:
-        (str): URL with removed brackets, if needed to remove, else the URL itself.
+        (str): URL with characters brackets, if needed to remove, else the URL itself.
     """
-    return url_[:-1] if url_[-1] in ['[', ']'] else url_
+
+    def is_ipv6(url: str) -> bool:
+        try:
+            # IPv6 should have square brackets around them, no need to remove
+            ipaddress.IPv6Address(re.findall(r"\[(.*?)]", url)[0])
+            return True
+
+        except (ipaddress.AddressValueError, IndexError):
+            return False
+
+    delimiters = {
+        "[": "]",
+        "(": ")",
+        "{": "}",
+        "'": "'",
+        '"': '"'
+    }
+
+    while url_[0] in delimiters and not is_ipv6(url_):
+        if delimiters.get(url_[0]) == url_[-1]:
+            url_ = url_[1:-1]
+        else:
+            url_ = url_[1:]
+
+    if is_ipv6(f'[{url_.split("//")[-1]}]'):
+        # In case we have an IPv6 without brackets, it's "invalid" but we will return it
+        return url_
+
+    if url_.count("/") < 3 and not is_ipv6(url_):
+        # If url has no path only letters are allowed in tld or port
+
+        while not url_[-1].isalpha() and not url_[-1].isnumeric():
+            url_ = url_[:-1]
+
+        last_part = url_.split('.')[-1]  # Get last part of the url
+
+        if ":" in last_part:
+            # Checks for port in URL to handle it as a special case
+            tld = last_part.split(":", 1)[0]
+            port = last_part.split(":", 1)[1]
+
+            if not port.isnumeric() or (not tld.isalpha() and not 1 <= int(tld) <= 255):
```

Comment:
?

Reply:
it's an edge-case of an IP as the host or a domain with a port.